docs/getting_started/go/hello_world_in_go/index.md

```diff
@@ -83,419 +85,778 @@ go get go.temporal.io/sdk
+
+Create the file `activities.go` in your project root:
+
+```command
+touch activities.go
 ```
 
-With your project workspace configured, you're ready to create your first Temporal Activity and Workflow. You'll start with the Workflow.
+Open the file `activities.go` in your editor and add the following code that imports dependencies and defines a struct to hold the Activities:
 
-## Create a Workflow
 
-Workflows are where you configure and organize the execution of Activities. You write a Workflow using one of the programming languages supported by a Temporal SDK. This code is known as a *Workflow Definition*. 
+<!--SNIPSTART go-ipgeo-activity-setup-->
+[activities.go](https://github.com/temporalio/temporal-tutorial-ipgeo-go/blob/main/activities.go)
+```go
+package iplocate
 
-In the Temporal Go SDK, a Workflow Definition is an [exported function](https://go.dev/tour/basics/3) with two additional requirements: it must accept `workflow.Context` as the first input parameter, and it must return `error`. Your Workflow function can optionally return another value, which you'll use to return the result of the Workflow Execution. You can learn more in the [Workflow parameters](https://docs.temporal.io/dev-guide/go/foundations#workflow-parameters) section of the Temporal documentation.
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+type HTTPGetter interface {
+	Get(url string) (*http.Response, error)
+}
+
+type IPActivities struct {
+	HTTPClient HTTPGetter
+}
+
+```
+<!--SNIPEND-->
+
+With the Go SDK, you can define Activities as regular Go functions, or you can create them as members of a struct. If you use a struct, you can then pass in shared objects like database connections or clients that all Activities can share. In this tutorial you'll pass an HTTP client into the struct. This will let you stub out the HTTP client implementation when you write tests for your Activities later.
+
+Now add the following code to define a Temporal Activity that retrieves your IP address from `icanhazip.com`:
+
+<!--SNIPSTART go-ipgeo-activity-ip-->
+[activities.go](https://github.com/temporalio/temporal-tutorial-ipgeo-go/blob/main/activities.go)
+```go
+// GetIP fetches the public IP address.
+func (i *IPActivities) GetIP(ctx context.Context) (string, error) {
+	resp, err := i.HTTPClient.Get("https://icanhazip.com")
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	ip := strings.TrimSpace(string(body))
+	return ip, nil
+}
+
+```
+<!--SNIPEND-->
+
+Whether you define Activities as exportable functions or as members of a struct, Activities can accept a `context` as their first argument. While this argument is optional for Activity Definitions, you should add it as you may need it for other Go SDK features as your application evolves. Additionally, Activities also need to return an `error`.
```

Comment:
Another clarity edit for the Go SDK that makes more sense to me but which you can take or leave:

```suggestion
With the Temporal Go SDK, both Activities and Workflows expect a Go `context` as their first argument. This argument is necessary to enable a number of other SDK features.

Like other Go functions, Activities should return an `error` as one of their arguments, so it can be checked after the Activity completes. This is because Go does not use a `try`/`catch` construct like other languages.
```